### ISSUE TITLE
Sold Price in Sales log is incorrect

### ISSUE THREAD

**Issue by echthard** [taniman/profit-trailer#1734]
I have no idea why the Sold Price is different than the values in Binance.

Sales log in PT
![image](https://user-images.githubusercontent.com/1479483/39444715-f3e81c18-4cb8-11e8-80c4-4cab06c8085f.png)

Trade History in Binance
![image](https://user-images.githubusercontent.com/1479483/39444775-2390a052-4cb9-11e8-95b0-68a695d340b5.png)

My 24h Order History in Binance
![image](https://user-images.githubusercontent.com/1479483/39444810-3bace51a-4cb9-11e8-803f-b7a24ca9772f.png)

**Comment by echthard** [comment]
@SgtBatten Is this really a duplicate of "incorrect bought price" ?This is about an incorrect sold price. I still have this issue. 

Example of what is happening:
When current price of a coin is 10, PT says to Binance to sell it against 9.2 and Binance afterwards says the average is 9.4, how is it possible that PT says the average is 9.63?

**Comment by SgtBatten** [comment]
Yes it is literally the exact same issue. Incorrect average (bought price) after partial sell.

Looks like you or someone opened a new one now anyway for 2.0.7

**Comment by echthard** [comment]
Ok, seems weird to me, bc PT does not need to calculate anything when selling a coin. It just needs to know from Binance what the average price was. 

I also think to know what PT might be doing wrong, bc how is it coming up with a price that is not communicated to Binance? I think it just takes the price of the highest buy order at that time in the orderbook. Even if that particular order is too low for my sell order. Bc of this, PT tries to sell it against a lower price and then ofc the average price is higher than that price. But PT cannot ever know what the average price was if it is not getting that price back from Binance. So if PT does not do that (obviously), then where comes the price from that PT says I have sold it against? It's neither the one that PT said it would sell to nor the average, it's a third price that is not recorded anywhere. So my best guess would be that PT is taking the price of the highest buy order at that exact moment.

And again, I think it's hard to believe this is the same issue. There is only one way to know what the average price is and that is by getting it from Binance. No calculation needs to be done, it does not make any sense to try to calculate.

**Comment by SgtBatten** [comment]
Hang on. Looking at your picture above it has the same bought price for both sales. And the sold price is the one that changed. Which is totally normal.

**Comment by SgtBatten** [comment]
Ohhh wait.. I think this is my mistake

**Comment by SgtBatten** [comment]
I'll clean this up sorry. Can you add buy order history prior to the sale to the opening post.

**Comment by echthard** [comment]
Sure. This is all I have for LINK:

![image](https://user-images.githubusercontent.com/1479483/40277709-458cedb2-5c24-11e8-8190-dc559745320d.png)

**Comment by echthard** [comment]
I can add more examples, bc I have this regularly. Just had one with KMD

![image](https://user-images.githubusercontent.com/1479483/40277737-a63fc6ca-5c24-11e8-813d-b189911f7cdc.png)

PT says it sold it at 0.0004179 while Binance says it sold it at 0.0004169 and the sell order was 0.0004162.

![image](https://user-images.githubusercontent.com/1479483/40277752-c446c952-5c24-11e8-8f96-31ba11184a18.png)

![image](https://user-images.githubusercontent.com/1479483/40277803-6173e566-5c25-11e8-896c-738cc3479154.png)

**Comment by echthard** [comment]
It's also the other way around. PT says it has sold MANA at 1443 sat, but Binance says it is 1448 sat.
